src/analysis/retail/priest/discipline/modules/guide/RaptureAnalysis.tsx

```diff
@@ -0,0 +1,290 @@
+  badCastIndexes?: number[];
+  damageRotation: CastEvent[];
+}
+
+class RaptureAnalysis extends Analyzer {
+  static dependencies = {
+    atonementModule: Atonement,
+    eventHistory: EventHistory,
+    globalCooldown: GlobalCooldown,
+    statTracker: StatTracker,
+    evangelism: Evangelism,
+    haste: Haste,
+  };
+
+  protected eventHistory!: EventHistory;
+  protected atonementModule!: Atonement;
+  protected globalCooldown!: GlobalCooldown;
+  protected statTracker!: StatTracker;
+  protected evangelism!: Evangelism;
+  protected haste!: Haste;
+
+  ramps: Ramp[] = [];
+  finishedRamping = false;
+  radianceCounter = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_PRIEST.RAPTURE_TALENT),
+      this.onRaptureCast,
+    );
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.buildSequence);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.fillDpsRotation);
+  }
+
+  // groups all the casts just before you cast evangelism
+  onRaptureCast(event: CastEvent) {
+    this.ramps.push({ timestamp: event.timestamp, rampHistory: [], damageRotation: [] });
+    console.log(this.ramps);
+    this.finishedRamping = false;
+    this.radianceCounter = 0;
+  }
+
+  // Need to build to the end of the ramp - be it the end of the rapture buff + two radiance casts, or if the buff is ended early by 2 radiances.
+  // If the sequence is too short, the damage rotation at the end will be cut by cleanupRamp().
+
+  buildSequence(event: CastEvent) {
+    if (this.ramps.length < 1) {
+      return;
+    }
+
+    if (
+      event.ability.guid === TALENTS_PRIEST.POWER_WORD_RADIANCE_TALENT.id &&
+      !this.finishedRamping
+    ) {
+      this.ramps[this.ramps.length - 1].rampHistory.push(event);
+      this.radianceCounter += 1;
+      return;
+    }
+
+    if (this.radianceCounter > 1) {
+      this.finishedRamping = true;
+      this.cleanupRamp();
+      return;
+    }
+
+    if (this.ramps[this.ramps.length - 1].timestamp + 12000 > event.timestamp) {
+      this.ramps[this.ramps.length - 1].rampHistory.push(event);
+    } else {
+      this.finishedRamping = true;
+      this.cleanupRamp();
+    }
+  }
+
+  // gets your spells cast 10s after pressing evangelism.
+  fillDpsRotation(event: CastEvent) {
+    if (this.ramps.length < 1 || !this.finishedRamping) {
+      return;
+    }
+
+    const lastRampCast = this.ramps[this.ramps.length - 1].rampHistory[
+      this.ramps[this.ramps.length - 1].rampHistory.length - 1
+    ];
+    if (event.timestamp < lastRampCast.timestamp + 10000) {
+      this.ramps[this.ramps.length - 1].damageRotation.push(event);
+    }
+  }
+
+  // edits the ramp history array to only include the applicators(or bad damage casts if there are damage casts in between)
+  cleanupRamp() {
+    let radCasted = false;
+    this.ramps[this.ramps.length - 1].rampHistory.forEach((rampCast, ix) => {
```

Comment:
rather than accessing `this.ramps[this.ramps.length - 1]` repeatedly, you should add a getter for that value, e.g.

```ts
get currentRamp() {
  return this.ramps[this.ramps.length - 1];
}
```

that way it is less repetitive and easier to alter if you change how you track the current ramp later.